### PR TITLE
fix: use separate endpoint for liveness probe (Kratos)

### PR DESCRIPTION
**Pull request by matthiastz**
This PR changes the default `livenessProbe`, because by default Kratos uses `/admin/health/ready` for both the `livenessProbe` and `readinessProbe` (see [link](https://github.com/ory/k8s/blob/5477f1955843a1666bc4f93fa868be5abef01d34/helm/charts/kratos/templates/deployment-kratos.yaml#L190-L213)). This setup leads to issues if you use Policy Engines like https://kyverno.io/ 

Example:
```
Warning  error  (REDACTED)  helm-controller  Helm install failed: admission webhook "validate.kyverno.svc-ignore" denied the request: 

policy Deployment/REDACTED/kratos for resource violation: 

require-pod-probes-to-be-different:
  validate-probes: Liveness and readiness probes cannot be the same.
```

We also recognized that Hydra uses two different endpoints for liveness/readiness, see: [link](https://github.com/ory/k8s/blob/5477f1955843a1666bc4f93fa868be5abef01d34/helm/charts/hydra/templates/deployment.yaml#L90-L113). Which is fine by default.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added the necessary documentation within the code base (if appropriate).

### CODE REVIEW

**Comment by tacurran** [comment]
@matthiastz Would you please add a more detailed use case for liveness probe?  We want to make sure additions to this repository are not aimed at use cases that present a higher security threat. Not all services should be started for instance without further investigation into root causes e.g. deadlock. Would you also add some additional information about why and how you will use this, along with some documentation?

File: helm/charts/kratos/templates/deployment-kratos.yaml [hunk 1]
```diff
@@ -192,7 +192,7 @@ spec:
             {{- toYaml .Values.deployment.customLivenessProbe | nindent 12 }}
           {{- else }}
             httpGet:
-              path: /admin/health/ready
+              path: /health/alive
```
Comment (scthi) [inline helm/charts/kratos/templates/deployment-kratos.yaml:195]: The default path for the admin api is `/admin` calling `/health/alive` results in a `307 Temporary Redirect` to `/admin/health/alive`.

@piotrmsc This pr solely switches to the correct alive probe endpoint [as documented](https://github.com/ory/kratos/blob/b75313e15d7071a77ac49a8425c1fac1be1f5bb3/.schema/openapi.json#L1366-L1444). There is no security aspect to it. 

```suggestion
              path: /admin/health/alive
```

**Review by scthi — CHANGES_REQUESTED** [review]

**Review by scthi — APPROVED** [review]

**Comment by tacurran** [comment]
@matthiastz - please address my comment above as part of the review, and add me as a reviewer. 
@matthiastz Would you please add a more detailed use case for liveness probe?

**Comment by scthi** [comment]
@tacurran This pr solely switches to the correct alive probe endpoint [as documented](https://github.com/ory/kratos/blob/b75313e15d7071a77ac49a8425c1fac1be1f5bb3/.schema/openapi.json#L1366-L1444). There is **no security aspect** to it. 
The technical reason why Liveness Probes should be different from Readiness Probes is explained here: https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html

**Review by Demonsthere — APPROVED** [review]
As we provide full customization for the probes i am going to merge this, as the previous behaviour can be easily restored :)

**Pull request merged**